WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2006,7 +2006,7 @@ private boolean isNewPost() {
     }
 
     private void saveResult(boolean saved, boolean uploadNotStarted) {
-        Intent i = getIntent();
+        Intent i = new Intent();
```

Comment:
By replacing `getIntent()` to `new Intent()` we'll be losing already existing intent extra data (e.g. `SITE`, `postModelLocalId`):

<img width="700" alt="intent extra" src="https://user-images.githubusercontent.com/1405144/173585456-d96b41cf-6655-45a6-962a-03d895785449.png">

which might be [needed by the calling activity's `onActivityResult()` ](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L1141-L1154).

Some of the [suggestions](https://developer.android.com/guide/components/intents-filters#UseIntentsMoreResponsibly) we can try:

- Copy only the essential extras within intents, and perform any necessary sanitation and validation.
   To the new `Intent`, you can copy intent extra data that might be needed by the calling activities. 

- Don't export your app's components unnecessarily. 
   Test that `EditPostActivity` is not invoked from any external process and mark `exported=false` in `AndroidManifest`.